### ISSUE TITLE
Build of 0.17.0-alpha.5- and 0.16.16- based project fails when targetting 32-bit Android

### ISSUE THREAD

**Issue by briansmith** [briansmith/ring#1106]
```
ld: error: /target/armv7-linux-androideabi/release/deps/libring-73b8f3cb0da57e3c.rlib(sha512-armv4-linux32.o): requires unsupported dynamic reloc R_ARM_REL32; recompile with -fPIC
ld: warning: shared library text segment is not shareable
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by briansmith** [comment]
I'm going to try resurrecting the `.comm` stuff I removed to see if that fixes it.

**Comment by briansmith** [comment]
WIP fix is PR #1109. I'm not sure if that will actually fix the problem and/or whether that will cause more problems though. Will be testing it more tomorrow.

**Comment by briansmith** [comment]
I was able to reduce this down to a minimum test case:

```
[package]
name = "so"
version = "0.1.0"
edition = "2018"

[lib]
crate-type = ["cdylib"]

[dependencies]
ring = "0.16.16"
```

`cat src/lib.rs`:
```
use ring::digest;

#[no_mangle]
pub unsafe fn foo() {
    let _ = digest::Context::new(&digest::SHA256);
}
```

The *ring* CI/CD succeeds because it never builds a shared library trying to use one of these features; it only builds executables, and this linking failure doesn't occur when building executables.

**Comment by briansmith** [comment]
Here's a test case for all four of the features I modified in the commit that caused this regression:
```
use ring::{aead, digest, signature};

#[no_mangle]
pub unsafe fn foo() {
    let _ = digest::Context::new(&digest::SHA256);
    let _ = digest::Context::new(&digest::SHA512);
    let _ = aead::UnboundKey::new(&aead::CHACHA20_POLY1305, &[0u8; 32]);
    signature::UnparsedPublicKey::new(&signature::RSA_PKCS1_2048_8192_SHA256, b"").verify(b"", b"").unwrap();
}
```

**Comment by briansmith** [comment]
I updated PR #1109 with an updated fix that I verified solves the build problem.

**Comment by briansmith** [comment]
Fixed in 0.16.17, which I've just published on crates.io.
